databroker/headersource/core.py

```diff
@@ -515,7 +515,12 @@ def insert_run_start(run_start_col, run_start_cache,
     col = run_start_col
     run_start = dict(time=time, uid=uid, **kwargs)
     apply_to_dict_recursively(run_start, sanitize_np)
-
+    try:
+        if run_start_given_uid(uid, run_start_col,
```

Comment:
I originally put it there then changed my mind. Do events ever get inserted via insert_one()? If not, it is definitely a cheap process.